packages/app-desktop/gui/MainScreen/commands/perNotebookSortOrder.ts

```diff
@@ -0,0 +1,212 @@
+import CommandService, { CommandContext, CommandDeclaration, CommandRuntime } from '@joplin/lib/services/CommandService';
+import Setting from '@joplin/lib/models/Setting';
+import { State } from '@joplin/lib/reducer';
+import app from '../../../app';
+import eventManager from '@joplin/lib/eventManager';
+import {
+	notesSortOrderFieldArray, NOTES_SORT_ORDER_SWITCH, SETTING_FIELD, SETTING_REVERSE, SETTING_PER_FIELD_REVERSAL_ENABLED,
+} from './notesSortOrderSwitch';
+import { _ } from '@joplin/lib/locale';
+
+
+export const PER_NOTEBOOK_SORT_ORDER = 'perNotebookSortOrder';
+export const SETTING_PER_NOTEBOOK_SORT_ORDER_ENABLED = 'notes.perNotebookSortOrderEnabled';
+
+let previousNotebookId: string = null;
+let ownSortOrders: { [key: string]: any } = null;
+const sharedSortOrder: { [key: string]: any } = {
+	field: 'user_updated_time',
+	reverse: true,
+	user_updated_time: true,
+	user_created_time: true,
+	title: false,
+	order: false,
+};
+
+export function selectedNotebookHasOwnSortOrder() {
+	return hasOwnSortOrder(getSelectedNotebookId());
+}
+
+export function hasOwnSortOrder(notebookId: string) {
+	return notebookId && ownSortOrders && ownSortOrders.hasOwnProperty(notebookId + SUFFIX_FIELD);
+}
+
+export const declaration: CommandDeclaration = {
+	name: PER_NOTEBOOK_SORT_ORDER,
+	label: () => _('Toggle own sort order'),
+};
+
+export const runtime = (): CommandRuntime => {
+	loadOwnSortOrders();
+	loadSharedSortOrder();
+	eventManager.appStateOn('notesParentType', onNotebookSelectionMayChange);
```

Comment:
In code, we use the term "folder", not "notebook", so please change to onFolderSelectionMayChange

Reply:
I see.

Reply:
finished. c723ebe